swim/member.go

```diff
@@ -73,6 +90,71 @@ func (m Member) checksumString(b *bytes.Buffer) {
 	b.WriteString(m.Address)
 	b.WriteString(m.Status)
 	b.WriteString(strconv.FormatInt(m.Incarnation, 10))
+	m.Labels.checksumString(b)
+}
+
+// checksumString adds the label portion of the checksum to the buffer that is
+// passed in. The string will not be appended in the case where labels are not
+// set on this member. This is for backwards compatibility reasons with older
+// versions.
+func (l LabelMap) checksumString(b *bytes.Buffer) {
```

Comment:
How about moving the label checksumming part to a different file? I think it's pure and quite independent?